### PR TITLE
NX Feat: Set a ID for devices - Issue#2543

### PR DESCRIPTION
**Pull request by LeonamMoura**
* **Please check if the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

* **What is the current behavior?** (You can also link to an open issue here)
before it was not possible to set an id for the device by GUI-NX

* **What is the new behavior (if this is a feature change)?**
It is now possible to set an ID of 2 to 6 mandatory hexadecimal characters

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
no

* **Is there any issue related to this PR in other repository?** (such as dojot/dojot)
dojot/dojot#2543

* **Other information**:

### CODE REVIEW

File: src/js/views/createDevice/steps/SummaryStep/SummaryStep.jsx [hunk 1]
```diff
@@ -167,6 +201,7 @@ const SummaryStep = ({
 
 SummaryStep.propTypes = {
```
Comment (LuanEdCosta) [inline src/js/views/createDevice/steps/SummaryStep/SummaryStep.jsx:206]: Missing `setDeviceId` prop type definition

File: src/js/views/createDevice/translations/pt_br.createDevice.i18n.json [hunk 2]
```diff
@@ -48,8 +48,12 @@
   "summaryStep": {
     "title": "Revise os modelos, certificados e dê um nome para o seu dispositivo",
     "hint": "Se notar que faltou alguma coisa é só voltar, corrigir e continuar!",
-    "deviceNameHint": "Estamos quase lá, pra finalizar, dê um nome ao seu dispositivo!",
+    "deviceNameHint": "Estamos quase lá, dê um nome ao seu dispositivo!",
     "deviceNamePh": "Nome do Dispositivo",
+    "deviceIdField": "Para finalizar, defina um Id para o seu dispositivo",
+    "deviceIdHint": "Opcional. Caso não seja preenchido, o Id do dispositivo será gerado automaticamente pela dojot.",
```
Comment (LuanEdCosta) [inline src/js/views/createDevice/translations/pt_br.createDevice.i18n.json]: You can remove the last part "pela dojot"

**Review by LuanEdCosta — CHANGES_REQUESTED** [review]
Some small details...

File: src/js/views/createDevice/View.jsx [hunk 3]
```diff
@@ -34,6 +34,7 @@ const CreateDevice = () => {
   const [selectedCertificate, setSelectedCertificate] = useState({});
   const [staticAttrValues, setStaticAttrValues] = useState({});
   const [deviceName, setDeviceName] = useState('');
+  const [deviceId, setDeviceId] = useState('');
```
Comment (luisfelipefurlan) [inline src/js/views/createDevice/View.jsx]: you can initialize the variable with null instead of ''

File: src/js/views/createDevice/View.jsx [hunk 4]
```diff
@@ -109,6 +110,7 @@ const CreateDevice = () => {
       actions.createDevice({
         fingerprint: selectedCertificate?.fingerprint || '',
         label: deviceName,
+        id: deviceId.trim() === '' ? null : deviceId,
```
Comment (luisfelipefurlan) [inline src/js/views/createDevice/View.jsx]: if you initialize the variable with null, this validation will not be necessary

File: src/js/views/createDevice/steps/SummaryStep/SummaryStep.jsx [hunk 5]
```diff
@@ -26,6 +28,10 @@ const SummaryStep = ({
     setDeviceName('');
   };
 
+  const handleClearDeviceId = () => {
+    setDeviceId('');
```
Comment (luisfelipefurlan) [inline src/js/views/createDevice/steps/SummaryStep/SummaryStep.jsx]: you can set it to null

**Review by luisfelipefurlan — CHANGES_REQUESTED** [review]

**Review by LuanEdCosta — DISMISSED** [review]
I agree with Luis, but I didn't find anything else to fix, so I'm approving.

**Review by LuanEdCosta — APPROVED** [review]

**Review by luisfelipefurlan — APPROVED** [review]

**Pull request merged**